engine/src/main/java/com/redhat/cloud/notifications/db/repositories/NotificationHistoryRepository.java

```diff
@@ -45,28 +45,21 @@ public void createNotificationHistory(NotificationHistory history) {
     /**
      * Update a stub history item with data we have received from the Camel sender
      *
-     * @param jo Map containing the returned data
-     * @return Nothing
      * @see com.redhat.cloud.notifications.events.FromCamelHistoryFiller for the source of data
      */
     @Transactional
     public boolean updateHistoryItem(NotificationHistory notificationHistory) {
-        String updateQuery = "UPDATE NotificationHistory SET details = :details, invocationResult = :result, status = :status, invocationTime = :invocationTime WHERE id = :id";
-        int count = statelessSessionFactory.getCurrentSession().createQuery(updateQuery)
+        String hql = "UPDATE NotificationHistory " +
+                "SET details = :details, invocationResult = :result, status = :status, invocationTime = :invocationTime " +
+                "WHERE id = :id";
+        int count = statelessSessionFactory.getCurrentSession().createQuery(hql)
                 .setParameter("details", notificationHistory.getDetails())
                 .setParameter("result", notificationHistory.isInvocationResult())
                 .setParameter("status", notificationHistory.getStatus())
                 .setParameter("id", notificationHistory.getId())
                 .setParameter("invocationTime", notificationHistory.getInvocationTime())
                 .executeUpdate();
-
-        if (count == 0) {
-            throw new NoResultException("Update returned no rows");
-        } else if (count > 1) {
-            throw new IllegalStateException("Update count was " + count);
-        }
```

Comment:
This could never happen since we're retrieving the records from the primary key.